configuration/pih/scripts/visit/encounterTypeConfig.js

```diff
@@ -1278,25 +1347,43 @@ angular.module("encounterTypeConfig", [])
         };
 
         return {
-          get: function(uuid, country, site) {
-            var encounterType = encounterTypes[uuid];
+          get: function(encounter, country, site) {
+
+            if (encounter == null || encounter.encounterType == null) {
+              return encounterTypes.DEFAULT;
+            }
+
+            let encounterType = encounterTypes[encounter.encounterType.uuid];
 
             if (encounterType == null) {
               return encounterTypes.DEFAULT;
             }
 
+            // resolve any country/site specific configs
             if (encounterType.hasOwnProperty(country)) {
-               if (encounterType[country].hasOwnProperty(site)) {
-                 return encounterType[country][site];
-               }
-               else {
-                 return encounterType[country].hasOwnProperty('DEFAULT') ?
-                   encounterType[country]['DEFAULT'] : encounterType[country];
-               }
+              if (encounterType[country].hasOwnProperty(site)) {
+                encounterType = encounterType[country][site];
+              } else {
+                encounterType = encounterType[country].hasOwnProperty('DEFAULT') ?
+                  encounterType[country]['DEFAULT'] : encounterType[country];
+              }
+            } else {
+              encounterType = encounterType.hasOwnProperty(['DEFAULT']) ?
+                encounterType['DEFAULT'] : encounterType;
+            }
+
+            // resolve any version-specific configs
+            if (encounterType.hasOwnProperty('versions')) {
+              if (encounter.form != null) {
+                encounterType = encounterType['versions'].hasOwnProperty(encounter.form.version) ?
+                  encounterType['versions'][encounter.form.version] : encounterType['versions']['DEFAULT']
+              }
+              else {
+                encounterType = encounterType['versions']['DEFAULT'];
+              }
             }
```

Comment:
Similarly,

```suggestion
            if (encounterType.versions) {
                encounterType = encounter.form && encounter.form.version ?
                    encounterType['versions'][encounter.form.version] || encounterType['versions']['DEFAULT'] :
                    encounterType['versions']['DEFAULT'];
            }
```

Reply:
Actually, consider whether you want to `console.error` something helpful in the case that `encounter.form && !encounter.form.version`, assuming that would imply a programming error in the calling code.